### PR TITLE
Expose additional JIT metrics

### PR DESCRIPTION
**Pull request by BruceForstall**
Add the following metrics:

- ActualCodeBytes - the actual number of generated code bytes.
- AllocatedHotCodeBytes - the number of bytes allocated for hot code.
- AllocatedColdCodeBytes - the number of bytes allocated for cold code.
- ReadOnlyDataBytes - the number of bytes allocated for read-only data
- GCInfoBytes - the number of bytes allocated (in the VM) for GC info; the GC info block size.
- EHClauseCount - the number of EH clauses reported to the VM.

Note that `AllocatedCodeBytes` might be less than `AllocatedHotCodeBytes + AllocatedColdCodeBytes` if misestimated the size of instructions that were generated.

Removed the special-cased SuperPMI "size" metric in favor of the new `ActualCodeBytes` metric. Fixed up superpmi.py to use the new metric.

### CODE REVIEW

**Comment by BruceForstall** [comment]
@jakobbotsch @dotnet/jit-contrib PTAL

File: src/coreclr/tools/superpmi/superpmi/superpmi.cpp [hunk 1]
```diff
@@ -159,12 +159,13 @@ static void PrintDiffsCsvRow(
...
     fw.PrintQuotedCsvField(baseRes.CompileResults->MethodFullName == nullptr ? "" : baseRes.CompileResults->MethodFullName);
     fw.Printf(
-        ",%s,%s,%s,%s,%s,%u,%u,%lld,%lld",
+        ",%s,%s,%s,%s,%s,%u,%u,%u,%u,%lld,%lld",
         baseRes.CompileResults->TieringName == nullptr ? "" : baseRes.CompileResults->TieringName,
         ResultToString(baseRes.Result), ResultToString(diffRes.Result),
         baseRes.IsMinOpts ? "True" : "False",
         hasDiff ? "True" : "False",
         baseRes.NumCodeBytes, diffRes.NumCodeBytes,
+        (uint32_t)baseRes.NumGCInfoBytes, (uint32_t)diffRes.NumGCInfoBytes,
```
Comment (jakobbotsch) [inline src/coreclr/tools/superpmi/superpmi/superpmi.cpp:168]: There is no need to add this manually, all metrics are automatically included

Comment (BruceForstall) [reply src/coreclr/tools/superpmi/superpmi/superpmi.cpp:168]: Maybe we should add hot/cold/total code size as metrics, then get rid of `NumCodeBytes` here (as well as getting rid of the `NumGCInfoBytes` that I added)?

Comment (jakobbotsch) [reply src/coreclr/tools/superpmi/superpmi/superpmi.cpp:168]: Yeah, we could definitely move more of this stuff into metrics now.

**Comment by BruceForstall** [comment]
@jakobbotsch I added more metrics (see updated top comment), and removed the special handling of the pre-existing "Size" metric. PTAL.

**Comment by BruceForstall** [comment]
Seems odd this is showing [TP diffs](https://dev.azure.com/dnceng-public/public/_build/results?buildId=851969&view=ms.vss-build-web.run-extensions-tab). Is the baseline run clean?

**Review by jakobbotsch — APPROVED** [review]
LGTM.

One subtle difference here will be that the code size will not be available anymore when using release JIT with superpmi.exe, since the JIT does not report metrics back to the VM in release. Not sure if anyone is using superpmi.py asmdiffs with release JIT though.

We can fix this by reporting back the metrics in release if some environment variable like `DOTNET_JitMetrics` is set. I've had that on my todo list for a while but haven't gotten around to it.

**Comment by BruceForstall** [comment]
> We can fix this by reporting back the metrics in release if some environment variable like DOTNET_JitMetrics is set.

I like the idea of optionally reporting metrics in Release.

**Pull request merged**